lib/msf/core/opt_rhost_url.rb

```diff
@@ -0,0 +1,75 @@
+# -*- coding: binary -*-
+
+module Msf
+  ###
+  #
+  # RHOST URL option.
+  #
+  ###
+  class OptRhostUrl < OptBase
+    def type
+      'rhost url'
+    end
+
+
+    def normalize(value)
+      return unless value
+
+      uri = get_uri(value)
+      option_hash = {}
+      # Blank this out since we don't know if this new value will have a `VHOST` to ensure we remove the old value
+      option_hash['VHOST'] = nil
+
+      option_hash['RHOSTS'] = uri.hostname
+      option_hash['RPORT'] = uri.port
+      option_hash['SSL'] = %w{ssl https}.include?(uri.scheme)
+
+      # Both `TARGETURI` and `URI` are used as datastore options to denote the path on a uri
+      option_hash['TARGETURI'] = uri.path || '/'
+      option_hash['URI'] = uri.path || '/'
+
+      if uri.scheme and %{http https}.include?(uri.scheme)
```

Comment:
```suggestion
      if uri.scheme && %[http https].include?(uri.scheme)
```

https://github.com/rapid7/metasploit-framework/pull/13766/files#r451880965